Lib/test/test_type_annotations.py

```diff
@@ -266,6 +267,17 @@ def check_annotations(self, f):
         f.__annotations__ = {"z": 43}
         self.assertIs(f.__annotate__, None)
 
+    def test_annotate_function_signature(self):
+        def f(x: int): pass
+        anno = f.__annotate__
+        self.assertIsInstance(anno, types.FunctionType)
+        self.assertEqual(f.__name__, "__annotate__")
+
+        expected_sig = inspect.Signature(
+            [inspect.Parameter("__format__", inspect.Parameter.POSITIONAL_ONLY)]
```

Comment:
``__format__`` is already used in a different context, as a method name. Can this be confusing?

Reply:
I feel it's unlikely to cause much confusion, since the name will very rarely show up to users (only if they introspect annotate functions, which is very unlikely to happen), and in a context that doesn't have anything to do with the `__format__` method.

Reply:
Given that we are kind of picking a name out of thin air that we expect not to matter, it seems like we might as well avoid the potential for someone thinking this is related to `__format__`? Would there be an issue with just using `_format`? (I don't feel strongly about this at all.)

Reply:
We should use a dundered name because dundered name are reserved to the implementation. Users could use a class named `_format` in their annotations.

The current PR has this behavior:
```
>>> def f(x: __format__): pass
... 
>>> f.__annotations__
{'x': 1}
```
I think with a dunder name we can handwave that away with "don't do that", but a user could reasonably use the name `_format`.

Still we could use a different name like `__fmt__` or `__annotate__` (I think Larry suggested the latter, but that name feels more confusing than `__format__`).

Reply:
Ah yeah, makes sense why it needs to be a dunder name. Given users should never have to type it, or likely see it, and the main thing we prefer to avoid is collisions with a user parameter, should we actually prefer something longer, like `__annotation_format__`?